### PR TITLE
Fix issue when Dockerfile present but BUILDPACK_URL is set

### LINKED ISSUE #1903: BUILDPACK_URL environment variable is ignored when Dockerfile is present
Hi all,

I'm experiencing an issues while trying to deploy [zendesk/samson](https://github.com/zendesk/samson) using buildpack instead of the bundled Dockerfile.

What I see is, that despite I have `BUILDPACK_URL: https://github.com/heroku/heroku-buildpack-ruby` set under the application, Dokku still tries to build the application with the bundled Dockerfile. [Gist](https://gist.github.com/cu12/9fbdc4ebe4d998df44a1)

As I can see in the [documentation](http://dokku.viewdocs.io/dokku/deployment/dockerfiles/) - point it out if I missed something - this is contrary to the desired behavior.

### PR DESCRIPTION
**Pull request by cu12**
Fixes  #1903

### CODE REVIEW

**Comment by josegonzalez** [comment]
I can't merge this as it doesn't use the config plugin to interact with the env file. Can you fix that?

**Comment by cu12** [comment]
Sure, I wasn't checking hard enough to find the proper way I guess ;)

**Comment by josegonzalez** [comment]
Search for `config_get` :)

**Comment by cu12** [comment]
It seems to be that CircleCI died during the test, but the linting was ok.

**Comment by josegonzalez** [comment]
@michaelshobbs should we have a test for this?

**Comment by michaelshobbs** [comment]
Probably

**Comment by josegonzalez** [comment]
@cu12 mind adding a test so we don't break this functionality in the future?

**Comment by cu12** [comment]
Sure, no problem.

**Comment by josegonzalez** [comment]
@cu12 *bump* :)

**Comment by cu12** [comment]
I started to cook up something for this one, but I only have minimal to none experience with bats. Hold on :)

**Comment by josegonzalez** [comment]
I leave the sense-making to @michaelshobbs (you'll notice I've never written a test for dokku ever).

File: tests/unit/10_build-env.bats [hunk 1]
```diff
@@ -44,6 +44,20 @@ teardown() {
   assert_success
 }
 
+@test "(build-env) buildpack if Dockerfile present" {
```
Comment (michaelshobbs) [inline tests/unit/10_build-env.bats]: I don't think this does what you think it does. Let me know what we're trying to do here and I'll slap something together.

Comment (cu12) [inline tests/unit/10_build-env.bats]: @michaelshobbs: My way of thinking here was, that we know that the `Dockerfile` application is NodeJS, so if I specify the PHP buildpack it should fail. 

There might be a better solution to this, so if you have anything in your mind, please let me know.

Comment (michaelshobbs) [inline tests/unit/10_build-env.bats]: The default app that `deploy_app` attempts to deploy does not contain a `Dockerfile` 
My understanding of this PR is to allow an app that contains a `Dockerfile` be deployed via buildpack if `BUILDPACK_URL` is set. In order to test that it seems like we should instead set `BUILDPACK_URL` to the nodejs buildpack, then call `deploy_app dockerfile` and test for success. Example:

```
@test "(build-env) buildpack deploy with Dockerfile" {
  run dokku config:set --no-restart $TEST_APP BUILDPACK_URL='https://github.com/heroku/heroku-buildpack-nodejs'
  echo "output: "$output
  echo "status: "$status
  assert_success

  run deploy_app dockerfile
  echo "output: "$output
  echo "status: "$status
  assert_success
}
```

Side note: Unless I'm missing something (which happens often depending on the day) this new test, does the same thing as the test below. Definitely correct me if I'm wrong there please. :smile: 

Also, apologies but you'll need to rebase this. We just did a big restructure for commands. I'm happy to take care of it if you'd prefer not to. Just let me know.

Comment (cu12) [inline tests/unit/10_build-env.bats]: If I understand this correctly, the test above will also be a success if Dokku builds `$TEST_APP` using `Dockerfile`, ignoring the fact that it should build using the buildpack set in `BUILDPACK_URL` instead. 

I'd be glad if you could rebase this as I'm pretty much slow to respond here due to a deadline :)

Comment (michaelshobbs) [inline tests/unit/10_build-env.bats]: Fair point about the test....I'll noodle on a better approach. Happy to rebase too. Now get back to "real" work :smile:

**Comment by michaelshobbs** [comment]
Closing in favor of rebased #1980

**Pull request closed without merging**